api/billingd/filrewards/filrewards.go

```diff
@@ -0,0 +1,217 @@
+package filrewards
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/textileio/textile/v2/api/billingd/analytics"
+	mdb "github.com/textileio/textile/v2/mongodb"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Event int
+
+const (
+	Unspecified Event = iota
+	FirstKeyAccountCreated
+	FirstKeyUserCreated // does this fit "register first user?"
+	FirstOrgCreated
+	InitialBillingSetup
+	FirstBucketCreated
+	FirstBucketArchiveCreated
+	FirstMailboxCreated
+	FirstThreadDbCreated
+)
```

Comment:
We need to keep a set of reward events that are specific to the FIL rewards code since they are used in a historical record, even if the associated `analytics` event no longer exists. These each have an associated `analytics.Event`.

Reply:
This also makes it easier to have other sources of events some day if we want to.

Reply:
nice